src/script/cond_stack.zig

```diff
@@ -0,0 +1,149 @@
+const std = @import("std");
+const OpConditional = @import("./opcodes/constant.zig").OpConditional;
+const testing = std.testing;
+const Allocator = std.mem.Allocator;
+
+/// Errors that can occur during stack operations
+pub const ConditionalStackError = error{
+    InvalidCondition,
+    DisabledOpcode,
+    UnbalancedConditional,
+};
+
+/// ConditionalStack represents a stack of conditional values used in script execution
+///
+/// This struct implements a stack data structure using a dynamic array.
+/// It manages conditional values (0, 1, 2) for branching in script execution.
+pub const ConditionalStack = struct {
+    const Self = @This();
+
+    /// Dynamic array to store conditional values
+    stack: std.ArrayList(u8),
+    /// Memory allocator used for managing stack storage
+    allocator: Allocator,
+
+    /// Initialize a new ConditionalStack
+    ///
+    /// Args:
+    ///     allocator: Memory allocator for managing stack storage
+    ///
+    /// Returns: A new ConditionalStack instance
+    pub fn init(allocator: Allocator) Self {
+        return .{
+            .stack = std.ArrayList(u8).init(allocator),
+            .allocator = allocator,
+        };
+    }
+
+    /// Deallocate all resources used by the ConditionalStack
+    pub fn deinit(self: *Self) void {
+        self.stack.deinit();
+    }
+
+    /// Push a conditional value onto the stack
+    ///
+    /// In Bitcoin script, conditional values are strictly limited to:
+    /// - OpConditional.False = 0
+    /// - OpConditional.True = 1
+    /// - OpConditional.Skip = 2
+    ///
+    /// Returns:
+    ///     Possible error if allocation fails
+    pub fn push(self: *Self, value: u8) !void {
+        try self.stack.append(value);
+    }
+
+    /// Pop a conditional value from the stack
+    pub fn pop(self: *Self) !void {
+        if (self.stack.items.len == 0) {
+            return ConditionalStackError.UnbalancedConditional;
+        }
+        _ = self.stack.pop();
```

Comment:
Don't you want to return the poped value?

Reply:
@tdelabro, I don't see any need to pop it is possible to remove the value completely. what do you think?